spanner/changestreams/src/main/java/com/example/spanner/changestreams/ChangeStreamSample.java

```diff
@@ -0,0 +1,255 @@
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.common.collect.ImmutableList;
+import com.google.spanner.admin.database.v1.UpdateDatabaseDdlMetadata;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sample code for querying change stream, it:
+ * 1. Creates a table with simple schema and a change stream that watches the table.
+ * 2. Inserts test data into the table.
+ * 3. Executes a change stream initial query to get change stream partition tokens.
+ * 4. Executes a change stream partition query to get data change records of the inserted rows.
+ * 5. Drops the created table and change stream.
+ */
+public class ChangeStreamSample {
+  private static final long TIMEOUT_MINUTES = 2;
+
+  public static void main(String... args) {
+    if (args.length != 3) {
+      System.err.println("Usage: ChangeStreamSample "
+          + "<instance_id> "
+          + "<database_id> "
+          + "<prefix>");
+      return;
+    }
+
+    final String instanceId = args[0];
+    final String databaseId = args[1];
+    final String prefix = args[2];
+    final String tableName = prefix + "_Singers";
+    final String changeStreamName = prefix + "_ChangeStreamSingers";
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseAdminClient dbAdminClient = spanner.getDatabaseAdminClient();
+    DatabaseClient dbClient = spanner.getDatabaseClient(
+        DatabaseId.of(options.getProjectId(), instanceId, databaseId));
+
+    try {
+      createChangeStream(dbAdminClient, instanceId, databaseId, tableName, changeStreamName);
+      queryChangeStream(dbClient, tableName, changeStreamName);
+      dropChangeStream(dbAdminClient, instanceId, databaseId, tableName, changeStreamName);
+    } catch (Exception e) {
+      System.out.println(e);
+    } finally {
+      spanner.close();
+    }
+  }
+
+  public static void updateDdl(
+      DatabaseAdminClient dbAdminClient, String instanceId, String databaseId,
+      List<String> statements) throws Exception {
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op = dbAdminClient
+        .updateDatabaseDdl(
+          instanceId,
+          databaseId,
+          statements,
+            null
+        );
+
+    op.get(TIMEOUT_MINUTES, TimeUnit.MINUTES);
+  }
+
+  // [START change_stream_sample_create_change_stream]
+  static void createChangeStream(
+      DatabaseAdminClient dbAdminClient, String instanceId, String databaseId, String tableName,
+      String changeStreamName) throws Exception {
+    System.out.println(
+        String.format("Updating database DDL to create table %s and change stream %s.",
+          tableName, changeStreamName));
+
+    updateDdl(
```

Comment:
nit: maybe it is just easier to inline this function so that the sample between tags is self-contained?

Reply:
Done.